camera_realsense.cpp

```
@@ -290,180 +332,395 @@ tuple<unsigned char*, size_t, bool> encodeDepthRAW(const unsigned char* data, co
+        std::shared_ptr<std::vector<uint16_t>> latestDepthFrame;
+        std::chrono::milliseconds latestTimestamp;
+        {
+            std::lock_guard<std::mutex> lock(GLOBAL_LATEST_FRAMES.mutex);
+            latestColorFrame = GLOBAL_LATEST_FRAMES.colorFrame;
+            latestDepthFrame = GLOBAL_LATEST_FRAMES.depthFrame;
+            latestTimestamp = GLOBAL_LATEST_FRAMES.timestamp;
+        }
+
+        for (const auto& sensor : this->props_.sensors) {
+            if (sensor == "color") {
+                std::unique_ptr<vsdk::Camera::raw_image> color_response;
+                color_response =
+                    encodeJPEGToResponse((const unsigned char*)latestColorFrame.get_data(),
+                                         this->props_.color.width, this->props_.color.height);
+                response.images.emplace_back(std::move(*color_response));
+            } else if (sensor == "depth") {
+                std::unique_ptr<vsdk::Camera::raw_image> depth_response;
+                depth_response = encodeDepthRAWToResponse(
+                    (const unsigned char*)latestDepthFrame->data(), this->props_.depth.width,
+                    this->props_.depth.height, this->props_.littleEndianDepth);
+                response.images.emplace_back(std::move(*depth_response));
+            }
+        }
+        response.metadata.captured_at =
+            std::chrono::time_point<long long, std::chrono::nanoseconds>(
+                std::chrono::duration_cast<std::chrono::nanoseconds>(latestTimestamp));
+        if (debug_enabled) {
+            auto stop = std::chrono::high_resolution_clock::now();
+            auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(stop - start);
+            std::cout << "[get_images]  total:           " << duration.count() << "ms\n";
+        }
+        return response;
+    }
+
+    vsdk::AttributeMap do_command(vsdk::AttributeMap command) override {
+        auto response =
```

Are we returning a shared pointer to an empty map? If so, why? If this is unimplemented, we could throw an error. If we need a dummy value, why not just return a unique pointer, the command argument so it echoes, or just a plain-old empty map?

---

Or you could do what you did for `get_point_cloud`

---

changed the response to print an error and return an empty AttributeMap (as was done in point_cloud)